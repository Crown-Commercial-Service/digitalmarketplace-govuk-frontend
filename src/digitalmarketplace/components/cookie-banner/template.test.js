const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('cookie-banner')

describe('cookie-banner', () => {
  it('default example passes accessibility tests', async () => {
    const $ = render('cookie-banner', examples.default)
    const results = await axe($.html())
    expect(results).toHaveNoViolations()
  })

  it('renders a cookie banner component with content and buttons', () => {
    const $ = render('cookie-banner', examples.default)
    expect($.html()).toMatchSnapshot()
  })

  it('renders a cookie banner component with a custom component ID', () => {
    const $ = render('cookie-banner', examples['with custom componentID'])
    const $component = $('.dm-cookie-banner')
    expect($component.attr('id')).toEqual('my-custom-component-id')
  })

  it('renders a cookie banner component with a custom service name', () => {
    const $ = render('cookie-banner', examples['with custom service name'])
    const cookieInfoLink = $('.dm-cookie-banner__link').html().trim()
    expect(cookieInfoLink).toEqual('How Digital Marketplace Admin uses cookies')
  })
})
