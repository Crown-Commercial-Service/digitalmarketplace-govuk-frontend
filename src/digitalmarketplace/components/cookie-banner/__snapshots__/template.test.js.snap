// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cookie-banner component renders a cookie banner with visible content and buttons 1`] = `
"<html><head></head><body><div id=\\"dm-cookie-banner\\" class=\\"dm-cookie-banner govuk-width-container\\" data-module=\\"dm-cookie-banner\\" role=\\"region\\" aria-label=\\"Cookie Banner\\" aria-describedby=\\"dm-cookie-banner__heading\\">
  <div class=\\"dm-cookie-banner__wrapper\\">
    <h2 class=\\"dm-cookie-banner__heading govuk-heading-m\\" id=\\"dm-cookie-banner__heading\\">
      Can we store analytics cookies on your device?
    </h2>
    <p class=\\"govuk-body\\">
      Analytics cookies help us understand how our website is being used.
    </p>
    <div class=\\"dm-cookie-banner__buttons\\">
      <button class=\\"govuk-button dm-cookie-banner__button dm-cookie-banner__button--accept\\" type=\\"submit\\" data-accept-cookies=\\"true\\" aria-describedby=\\"dm-cookie-banner__heading\\" data-module=\\"govuk-button\\">
        Yes<span class=\\"govuk-visually-hidden\\">, Digital Marketplace can store analytics cookies on your device</span>
      </button>
      <button class=\\"govuk-button dm-cookie-banner__button dm-cookie-banner__button--reject\\" type=\\"submit\\" data-accept-cookies=\\"false\\" aria-describedby=\\"dm-cookie-banner__heading\\" data-module=\\"govuk-button\\">
        No<span class=\\"govuk-visually-hidden\\">, Digital Marketplace cannot store analytics cookies on your device</span>
      </button>
      <a class=\\"govuk-link dm-cookie-banner__link\\" href=\\"/cookies\\">How Digital Marketplace uses cookies</a>
    </div>
  </div>

  <div class=\\"dm-cookie-banner__confirmation\\" tabindex=\\"-1\\">
    <p class=\\"dm-cookie-banner__confirmation-message govuk-body\\">
      You can <a class=\\"govuk-link\\" href=\\"/user/cookie-settings\\">change your cookie settings</a> at any time.
    </p>
    <button class=\\"dm-cookie-banner__hide-button govuk-link\\" data-hide-cookie-banner=\\"true\\" role=\\"link\\" data-module=\\"govuk-button\\">Hide</button>
  </div>
</div>
</body></html>"
`;
